src/main/java/entities/WordFactory.java

```diff
@@ -25,10 +26,10 @@ public WordFactory(ValidityChecker v) {
      * @throws InvalidWordException if its string representation is not valid
      */
     public Word create(String word, Player author) throws InvalidWordException {
-        if(validityChecker.isValid(word))
-            return new Word(word, author);
-        else
+        String checkedWord = validityChecker.isValid(word);
+        if(checkedWord == null)
             throw new InvalidWordException("The word is not valid!");
+        return new Word(word, author);
```

Comment:
Don't you want to pass in the checkedWord into the Word constructor, the one that was potentially "corrected" by the validity checker?